blog/02-18-2023-Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain/Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain.md

```diff
@@ -0,0 +1,998 @@
+     }
+```
+
+The `UpgradeNFT()` function allows a player to upgrade the power value of their NFT. This function takes an `index` as a parameter which is used to specify the NFT to be upgraded. The function requires that the sender of the transaction is the owner of the NFT. If this requirement is met, the function transfers 0.5 Celo from the owner of the nft to the contract owner and increases the power value of the NFT and the player. 
+
+```solidity
+ function canSwallowNFT(address _address, uint _index) public view returns(bool){
+        if(playerpowervalue[_address] > playerpowervalue[nfts[_index].owner]){
+            return true;
+        }else{
+            return false;
+        }
+    }
+```
+
+The `CanSwallowNFT()` function returns true if the power value of the attacker is greater than the power value of the owner of the NFT. The function takes an `address` and an `index` as parameters. The address is used to specify the attacker and the index is used to specify the NFT to be swallowed. The function returns true if the power value of the attacker is greater than the power value of the owner of the nft. 
+
+```solidity
+ function hasMinted(address _address) public view returns(bool){
+        if(minters[_address] == true){
+            return true;
+        }else{
+            return false;
+        }
+    }
+```
+
+The `HasMinted()` function returns true if the address has minted an NFT. The function takes an `address` as a parameter which is used to specify the address to check. The function returns true if the address has minted an NFT. This will be usefull in our front-end.
+
+```solidity
+ function getAllNFTS(uint _index) public view returns(NFT memory){
+        return nfts[_index]; 
+    }
+```
+
+The `GetAllNFTS()` function returns all NFTs. The function takes an `index` as a parameter which is used to specify the NFT to be returned. The function returns an NFT struct with the given `index`.
+
+```solidity
+ function remove(uint _index) external {
+	        require(msg.sender == nfts[_index].owner, "can't remove this nft");         
+            nfts[_index] = nfts[allNFTs - 1];
+            delete nfts[allNFTs - 1];
+            allNFTs--; 
+            _transfer(address(this), msg.sender, nfts[_index].tokenId);
+	 }
+```
+
+The `Remove()` function removes an NFT from the game. The function takes an `index` as a parameter which is used to specify the NFT to be removed. The function requires that the sender of the transaction is the owner of the NFT. If this requirement is met, the function transfers the NFT from the contract to the owner and deletes the NFT from the internal nfts mapping. 
+
+```solidity
+ function getNFTlength() public view returns (uint256) {
+        return allNFTs;
+    }
+```
+
+The `GetNFTlength()` function returns the length of the NFTs in the list. The function returns the value of the `allNFTs` variable which is the length of all monster NFTs in the list. 
```

Comment:
The `GetNFTlength()` function returns the length of the NFTs in the list. The function returns the value of the `allNFTs` variable, which is the length of all monster NFTs in the list.